libs/java/service/src/main/java/ca/bc/gov/tno/services/data/TnoApi.java

```diff
@@ -0,0 +1,218 @@
+package ca.bc.gov.tno.services.data;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.FormHttpMessageConverter;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+import ca.bc.gov.tno.services.models.ContentReference;
+import ca.bc.gov.tno.services.models.DataSource;
+import ca.bc.gov.tno.services.data.config.ApiConfig;
+import ca.bc.gov.tno.services.models.TokenResponse;
+
+/**
+ * TnoApi class, provides helper methods for the managing the datasource.
+ */
+@Service
+public final class TnoApi {
+  /** Configuration settings for the api */
+  protected final ApiConfig config;
+
+  /** AJAX Requester */
+  protected final RestTemplate rest;
+
+  /** Java Web Token response from keycloak */
+  private TokenResponse token;
+
+  /**
+   * JSON content type that support charset=utf-8.
+   */
+  private final MediaType jsonMediaType;
+
+  /**
+   * Creates a new instance of a TnoApi object.
+   * 
+   * @param config API configuration settings.
+   */
+  @Autowired
+  public TnoApi(ApiConfig config) {
+
+    this.config = config;
+    this.rest = new RestTemplate();
+
+    var pmap = new HashMap<String, String>(4);
+    pmap.put("charset", "utf-8");
+    this.jsonMediaType = new MediaType("application", "json", pmap);
+
+    var xml = new MappingJackson2HttpMessageConverter();
+    xml.setSupportedMediaTypes(Arrays.asList(new MediaType[] { MediaType.TEXT_XML }));
+    var json = new MappingJackson2HttpMessageConverter();
+    json.setSupportedMediaTypes(
+        Arrays.asList(
+            new MediaType[] { MediaType.APPLICATION_JSON, jsonMediaType }));
```

Comment:
I have explicitly told it to accept `application/json?charset=utf-8` but it doesn't work...